### PR TITLE
feat(builder): export core build

### PR DESCRIPTION
**Pull request by Asuka109**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/builder-webpack-provider/src/exports/webpack.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 // eslint-disable-next-line @typescript-eslint/no-restricted-imports
 import webpack from 'webpack';
 
+export type { BuildExecuter, WebpackBuildError } from '../core/build';
```
Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/src/exports/webpack.ts:4]: webpack 这个文件主要是为了导出 webpack 自身的模块

像这两个方法，可以从主入口导出

Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/src/exports/webpack.ts:4]: 导出 BuildExecuter 的使用场景是？

Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/exports/webpack.ts:4]: okk

Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/exports/webpack.ts:4]: 暂时没有明确场景 我给去了吧

File: packages/builder/builder-webpack-provider/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 export { builderWebpackProvider } from './provider';
+export { webpackBuild } from './core/build';
```
Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/src/index.ts:2]: 那导出 webpackBuild 的使用场景是啥

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**